### ISSUE TITLE
Changing SharpRaven Data before Capture

### ISSUE THREAD

**Issue by MikeDrewitt** [getsentry/raven-csharp#206]
Currently I'm looking for a way to change the request data before it is sent to sentry. I'm trying to do this because the API request that's coming in is larger than 100Kb which Sentry doesn't accept. 

My original thought was to just change the Data before the capture removing the large field, but the Data is set to readonly. My second thought was to create a new JsonPacketFactory in the BeforeSend method on the RavenClient and create a new requester all together, but in the current 2.2.0 release the constructor for Requester is internal. I see that this was recently changed in a commit but has yet to be released in an official version.  

So in short my issue, is there any way currently to change the request data before capture that I'm missing? Or is this a situation of just waiting until the next release and utilize the new public Requester constructor?

**Comment by asbjornu** [comment]
Can you please have a look at #212 and see if the discussion there is of any help?

**Comment by debrandt** [comment]
I was having the same question. Found out that the JsonPacket does implement setters for its properties. You can access the JsonPacket through the Packet property of the requester.

**Comment by MikeDrewitt** [comment]
@debrandt Thanks for letting me know. I had figured out a different solution to the problem I was fixing, but looking into your solution that would have fixed my issue.

**Comment by asbjornu** [comment]
@MikeDrewitt: Does that mean this issue can be closed?

**Comment by bruno-garcia** [comment]
We've released the new SDK which solves this issue:

https://github.com/getsentry/sentry-dotnet/releases

https://www.nuget.org/packages/Sentry

With the new SDK there are two ways to go about this. One is using a `Func<SentryEvent, SentryEvent>` called `BeforeSend`, which allows you modify the event, avoid sending the event altogether (returning null) or simply returning a completely new event. 

For example:

```csharp
SentrySdk.Init(o => 
{
  // Drop an event altogether:
  if (@event.Tags.ContainsKey("SomeTag"))
  {
    return null;
  }
});
```

Another way is using a `ExceptionProcessor`. You create a [class implementing an interface](https://github.com/getsentry/sentry-dotnet/blob/e96be9fab28111d154b5431e28f5d069f30937fe/samples/Sentry.Samples.Console.Customized/Program.cs#L212-L224), and [add an instance to the SDK](https://github.com/getsentry/sentry-dotnet/blob/e96be9fab28111d154b5431e28f5d069f30937fe/samples/Sentry.Samples.Console.Customized/Program.cs#L112).

Also, note that the new SDK `gzip` compresses the body which means you can get around sending a much bigger event to Sentry. Perhaps because of that you won't even need to remove data from your events.

I'll close this issue but please feel free to reopen or open one in the new repository if you have any issues integrating. We'll be happy to help you moving over to the new SDK.
